pennylane_lightning/src/simulator/Measures.hpp

```diff
@@ -309,6 +310,76 @@ class Measures {
         return expected_value_list;
     };
 
+    size_t mcmc_step(const SVType &sv,
+                     std::unique_ptr<TransitionKernel<fp_t>> &tk,
+                     std::mt19937 &gen,
+                     std::uniform_real_distribution<fp_t> &distrib, size_t s1) {
+        auto s1_plog =
+            log((sv.getData()[s1] * std::conj(sv.getData()[s1])).real());
+
+        auto s1_qratio = tk->operator()(s1);
+
+        // transition kernel outputs these two
+        auto &s2 = s1_qratio.first;
+        auto &qratio = s1_qratio.second;
+
+        auto s2_plog =
+            log((sv.getData()[s2] * std::conj(sv.getData()[s2])).real());
+
+        auto alph = std::min(1., qratio * exp(s2_plog - s1_plog));
```

Comment:
```suggestion
        auto alph = std::min<fp_t>(1., qratio * exp(s2_plog - s1_plog));
```

The failures seem to be related to problems with argument type deduction. Explicitly giving the templated type here most likely will fix the Mac and Windows failures.

Reply:
Thank you Lee for the suggestion